### PR TITLE
fix: call create_end for all code paths

### PR DESCRIPTION
**Pull request by Wodann**

### CODE REVIEW

File: crates/revm/src/evm_impl.rs [hunk 1]
```diff
@@ -554,12 +554,23 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> EVMImpl<'a, GSPEC, DB,
...
+                            bytes,
+                        )
+                    } else {
+                        (
+                            InstructionResult::CreateContractSizeLimit,
+                            ret,
+                            interpreter.gas,
+                            bytes,
+                        )
+                    };
```
Comment (rakita) [inline crates/revm/src/evm_impl.rs:573]: This is not exactly the pattern used elsewhere

https://github.com/bluealloy/revm/blob/5d6ecd03ca8cf749b5f0d38a95fa628c32fc8387/crates/revm/src/evm_impl.rs#L405-L412

by default `create_end` would return `::Continue`

Comment (Wodann) [reply crates/revm/src/evm_impl.rs:573]: Isn't Inspector::create_end supposed to be called for every possible reason the create function might end?

Comment (Wodann) [reply crates/revm/src/evm_impl.rs]: > by default `create_end` would return `::Continue`

And doesn't the default implementation of `create_end` return `ret`?

https://github.com/bluealloy/revm/blob/5d6ecd03ca8cf749b5f0d38a95fa628c32fc8387/crates/revm/src/inspector.rs#L123

**Comment by Wodann** [comment]
I made sure `create_end` is now called for all return code paths

**Review by rakita — APPROVED** [review]
nice pattern!

**Pull request merged**